clash-lib/src/Clash/Driver/Manifest.hs

```diff
@@ -65,7 +72,58 @@ data Manifest
   , fileNames :: [(FilePath, ByteString)]
     -- ^ Names and hashes of all the generated files for the @TopEntity@. Hashes
     -- are SHA256.
-  } deriving (Show,Read)
+  } deriving (Show,Read,Eq)
+
+instance ToJSON Manifest where
+  toJSON (Manifest{..}) =
+    Aeson.object
+      [ "version" .= ("unstable" :: Text)
+      , "hash" .= manifestHash
+      , "flags" .= successFlags
+        -- TODO: add nested ports (i.e., how Clash split/filtered arguments)
+      , "components" .= componentNames
```

Comment:
If there is a manifest for each entity directory, when would components ever be more than one thing? I'm slightly unsure what this field is anyway

Reply:
Right makes sense, I didn't realise we called these components. Is there any benefit in keeping this field then, it seems to just be the information from `files` but without the hashes?

Reply:
Hmmyeah. In _theory_ the filenames can be wildly different from component names. Clash itself doesn't really generate such files, but it might happen once we do some magic with vendor IP?